pkg/operation/botanist/component/kubeapiserver/deployment.go

```diff
@@ -724,181 +722,17 @@ func (k *kubeAPIServer) handleVPNSettings(
 	secretCAVPN *corev1.Secret,
 	secretHTTPProxy *corev1.Secret,
 	secretLegacyVPNCAClient *corev1.Secret,
-	secretLegacyVPNSeed *corev1.Secret,
-	secretLegacyVPNSeedTLSAuth *corev1.Secret,
 	secretHAVPNSeedClient *corev1.Secret,
 	secretHAVPNSeedClientSeedTLSAuth *corev1.Secret,
 ) {
-	if !k.values.VPN.ReversedVPNEnabled {
-		k.handleVPNSettingsLegacy(deployment, secretLegacyVPNCAClient, secretLegacyVPNSeed, secretLegacyVPNSeedTLSAuth)
-	} else if k.values.VPN.HighAvailabilityEnabled {
+
+	if k.values.VPN.HighAvailabilityEnabled {
 		k.handleVPNSettingsHAReversedVPN(deployment, secretCAVPN, secretHAVPNSeedClient, secretHAVPNSeedClientSeedTLSAuth)
 	} else {
 		k.handleVPNSettingsReversedVPN(deployment, configMapEgressSelector, secretCAVPN, secretHTTPProxy)
 	}
 }
 
-func (k *kubeAPIServer) handleVPNSettingsLegacy(
-	deployment *appsv1.Deployment,
-	secretLegacyVPNCAClient *corev1.Secret,
-	secretLegacyVPNSeed *corev1.Secret,
-	secretLegacyVPNSeedTLSAuth *corev1.Secret,
-) {
-	deployment.Spec.Template.Labels[v1beta1constants.LabelNetworkPolicyToShootNetworks] = v1beta1constants.LabelNetworkPolicyAllowed
-	deployment.Spec.Template.Spec.InitContainers = []corev1.Container{{
-		Name:  "set-iptable-rules",
-		Image: k.values.Images.AlpineIPTables,
-		Command: []string{
-			"/bin/sh",
-			"-c",
-			"iptables -A INPUT -i tun0 -p icmp -j ACCEPT && iptables -A INPUT -i tun0 -m state --state NEW -j DROP",
-		},
-		SecurityContext: &corev1.SecurityContext{
-			Capabilities: &corev1.Capabilities{
-				Add: []corev1.Capability{"NET_ADMIN"},
-			},
-			Privileged: pointer.Bool(true),
-		},
-		VolumeMounts: []corev1.VolumeMount{{
-			Name:      volumeNameLibModules,
-			MountPath: volumeMountPathLibModules,
-		}},
-	}}
-	deployment.Spec.Template.Spec.Volumes = append(deployment.Spec.Template.Spec.Volumes, corev1.Volume{
-		Name: volumeNameLibModules,
-		VolumeSource: corev1.VolumeSource{
-			HostPath: &corev1.HostPathVolumeSource{Path: "/lib/modules"},
-		},
-	})
-
-	vpnSeedContainer := corev1.Container{
-		Name:            containerNameVPNSeed,
```

Comment:
There is no such container named `vpn-client`. Can you elaborate why there is HVPA config for this container https://github.com/gardener/gardener/blob/75423d6653b6e3328a1660873718fc322bc14bdf/pkg/operation/botanist/component/kubeapiserver/hvpa.go#L75-L79 and dedicated parser https://github.com/gardener/gardener/blob/78e6b41ebc3f2430ba5e8cd4ebbea6a78e2f84c2/pkg/operation/botanist/component/kubeapiserver/logging.go#L49-L54

Reply:
I'm wondering, if we should add the additional containers for the HA case here (vpn-client-0, vpn-client-1, vpn-path-controller)
@MartinWeindel